### ISSUE TITLE
Enrollments and groups

### ISSUE THREAD

**Issue by BuddyEthridge** [lsuits/lsu-block_simple_restore#21]
Currently, Simple restore wipes out enrollments and groups and then rebuilds them at the end of the process.  This removes all manually added users.  The "course specific settings" in Simple Restore are set to:

Keep current roles and enrollments: No
Keep current groups and grouping: No
Overwrite course configuration: Yes

The desired behavior would be for these to all be set as Yes, but Robert says that there is a bug that causes Yes, Yes, Yes to not work.  Please look into this when time permits.

**Comment by philcali** [comment]
Hmmm, the may have been a bug in the past, but I don't see how there would be a problem currently. CPS handles a simple restore event and listens to the aforementioned settings, here: https://github.com/lsuits/cps/blob/master/events/simple_restore.php#L35.

After giving this some more thought, I actually think it's a Moodle core bug (or behavior) that does something unexpected to course materials if those options are selected. I need to be refreshed.

Regardless, simple restore completely ignores the "Overwrite course config setting" (with CPS installed), as it will instead restore the course's previous settings anyway.

**Comment by rrusso** [comment]
Ok, the issue is that we are not keeping current enrollments.

CPS handles the event and re-enrolls UES students, but **NOT** manual enrollments.

If we change settings to **yes** **yes** **yes**, things break...badly.

**Comment by philcali** [comment]
What I don't understand is this: why doesn't **yes**, **yes**, **whatever** work? Does the underlying regular restore behave differently with the keep enrollment settings (other wiping enrollment or not)?

I don't understand how this is a simple restore issue, because this can be changed administratively.

**Comment by rrusso** [comment]
Therein lies the issue.

Yes yes whatever works in Moodle core, but freaks in simple restore for some reason.

At least it used to. Haven't tested recently.

**Comment by philcali** [comment]
The only thing it does now is some house keeping: https://github.com/lsuits/cps/blob/master/events/simple_restore.php#L35

For whatever reason, I noticed that the enrollment methods were being duped in a course, and I manually deleted one. This would cause things like two reprocess links, and other weird stuff happening in the course edit form. Anything else, it affects like the regular restore.

**Comment by rrusso** [comment]
I'll test tomorrow....

**Comment by philcali** [comment]
@rrusso, do you get a chance to test this?

**Comment by rrusso** [comment]
No, can you please?

**Comment by philcali** [comment]
OK, here's what I found with everything enabled:
1. Restores happen very quickly
2. All of the previous enrollments were in-tact.
3. It dupes the Guest Access enrollment method, but I think that's an underlying restore problem.

The only nuance I noticed about the duped Guest Access enrollments are how it shows up twice in the course edit form. If this bothers the instructor, they can simply remove it.

What problems were you experiencing before?
